calicoctl/commands/node/diags.go

```diff
@@ -155,7 +160,49 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getNodeContainerLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getNodeContainerLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	// Get a list of Calico containers running on this Node.
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}: {{.CreatedAt}}")
+	cmd.Stderr = &stderr
+	containers, err := cmd.Output()
+	if err != nil {
+		fmt.Printf("Could not run docker command: %s\n", stderr.String())
+		return
+	}
+
+	// No Calico containers found.
+	if string(containers) == "" {
+		log.Debug("Did not find any Calico containers")
+		return
+	}
+
+	fmt.Println("Copying logs from Calico containers")
```

Comment:
Some output I got from running on a k8s system:

```
$ docker ps -a --filter name=calico --format "{{.Names}}: {{.CreatedAt}}"
k8s_install-cni_calico-node-6p0xp_kube-system_603c4dd8-98c8-11e7-acf3-42010a800008_0: 2017-09-13 21:13:44 +0000 UTC
k8s_calico-node_calico-node-6p0xp_kube-system_603c4dd8-98c8-11e7-acf3-42010a800008_0: 2017-09-13 21:13:36 +0000 UTC
k8s_calico-etcd_calico-etcd-t6dwl_kube-system_5fe749f4-98c8-11e7-acf3-42010a800008_0: 2017-09-13 21:13:26 +0000 UTC
k8s_POD_calico-node-6p0xp_kube-system_603c4dd8-98c8-11e7-acf3-42010a800008_0: 2017-09-13 21:13:23 +0000 UTC
k8s_POD_calico-etcd-t6dwl_kube-system_5fe749f4-98c8-11e7-acf3-42010a800008_0: 2017-09-13 21:13:23 +0000 UTC
```

We should filter out anything with `k8s_POD_` in it since those are just the infra containers and won't have useful logs.

Reply:
Ah, did not know that, will filter them out.